lib/elixir/lib/module.ex

```diff
@@ -1099,10 +1099,23 @@ defmodule Module do
   ## Helpers
 
   defp preprocess_attribute(key, value) when key in [:moduledoc, :typedoc, :doc] do
-    unless match?({line, _} when is_integer(line), value) do
-      raise ArgumentError, "expected #{key} attribute given in the {line, doc} format, got: #{inspect(value)}"
+    case value do
+      {line, doc} when is_integer(line) and (is_binary(doc) or is_boolean(doc) or is_nil(doc)) ->
+        value
+      {line, doc} when is_integer(line) ->
+        # Here, either the user used "@moduledoc :not_a_binary" or
+        # "Module.put_attribute(..., {1, :not_a_binary})". By showing just the
+        # "doc" value in the error, it should be clear in both cases.
+        raise ArgumentError,
+          "expected the #{key} attribute to be a binary, a boolean, or nil, got: #{inspect(doc)}"
```

Comment:
what if we cheat here and say "to contain" instead of "to be". Its dual meaning implies that: "i expected the tuple to contain one of those" OR "i expected the attribute to contain one of those".